src/core/mac/mac_frame.cpp

```diff
@@ -970,25 +972,130 @@ const TimeIe *Frame::GetTimeIe(void) const
 }
 #endif // OPENTHREAD_CONFIG_TIME_SYNC_ENABLE
 
+#if OPENTHREAD_CONFIG_MULTI_RADIO
+uint16_t Frame::GetMtu(void) const
+{
+    uint16_t mtu;
+
+    switch (GetRadioType())
+    {
+#if OPENTHREAD_CONFIG_RADIO_LINK_IEEE_802_15_4_ENABLE
+    case kRadioTypeIeee802154:
+        mtu = OT_RADIO_FRAME_MAX_SIZE;
+        break;
+#endif
+
+#if OPENTHREAD_CONFIG_RADIO_LINK_TREL_ENABLE
+    case kRadioTypeTrel:
+        mtu = Trel::Link::kMtuSize;
+        break;
+#endif
+
+#if OPENTHREAD_CONFIG_RADIO_LINK_TOBLE_ENABLE
+    case kRadioTypeToble:
+        mtu = Toble::Link::kMtuSize;
+        break;
+#endif
+
+    default:
+        OT_ASSERT(false);
+    }
+
+    return mtu;
+}
+
+uint8_t Frame::GetFcsSize(void) const
+{
+    uint8_t fcsSize;
+
+    switch (GetRadioType())
+    {
+#if OPENTHREAD_CONFIG_RADIO_LINK_IEEE_802_15_4_ENABLE
+    case kRadioTypeIeee802154:
+        fcsSize = k154FcsSize;
+        break;
+#endif
+
+#if OPENTHREAD_CONFIG_RADIO_LINK_TREL_ENABLE
+    case kRadioTypeTrel:
+        fcsSize = Trel::Link::kFcsSize;
+        break;
+#endif
+
+#if OPENTHREAD_CONFIG_RADIO_LINK_TOBLE_ENABLE
+    case kRadioTypeToble:
+        fcsSize = Toble::Link::kFcsSize;
+        break;
+#endif
+
+    default:
+        OT_ASSERT(false);
+    }
+
+    return fcsSize;
+}
+
+#elif !OPENTHREAD_CONFIG_RADIO_LINK_IEEE_802_15_4_ENABLE
+
+uint16_t Frame::GetMtu(void) const
+{
+    return
+#if OPENTHREAD_CONFIG_RADIO_LINK_TREL_ENABLE
+        Trel::Link::kMtuSize;
+#elif OPENTHREAD_CONFIG_RADIO_LINK_TOBLE_ENABLE
+        Toble::Link::kMtuSize;
+#endif
```

Comment:
This is actually intentional. We want the `GetMtu()` to be inlined and threrefore defined in the header file in the simple and common case where we only have 15.4 radio link (this is to avoid code size overhead in the simple case).

Note that  moving the implementation to header in general case is not as easy/straight-forward since it creates a header dependency loop:
- We would need to include `trel_link.hpp` in `mac_frame.hpp` to see the `Trel::Link::kMtuSize` constant
- At the same time `trel_link.hpp` would need to see full definitions of `Mac::RxFrame` and `Mac::TxFrame` so it needs to include `mac_frame.hpp`.